torch/_torch_docs.py

```diff
@@ -790,8 +791,18 @@ def merge_dicts(*dicts):
 
 Convert the data into a `torch.Tensor`. If the data is already a `Tensor` with the same `dtype` and `device`,
```

Comment:
Let's just update this doc a bit while we're here:

"Converts :attr:\`data\` into a tensor. 

If :attr:\`data\` is already a tensor then this is equivalent to `data.to(device, dtype)`.

This operator also supports scalars, iterables, and objects that implement the CPU Array Interface or the CUDA Array Interface. CPU tensors constructed an object implementing the CPU Array Interface, like a NumPy array, share their storage with that object, unless a different :attr:\`dtype\` is requested. Similarly, CUDA tensors constructed from an object implementing the CUDA Array Interface share their storage with that object, unless a different :attr:\`dtype\` is requested. In all other cases the values in :attr:\`data\` are copied."

What would you think of something like this? I actually have some follow-up questions:

- is this actually equivalent to .to(device, dtype) when data is a tensor? I think it is.
- are there objects outside of scalars, iterables and {CPU, CUDA} Array Interface objects this supports?

Reply:
Answers:
- yes to the `to` equivalence question, see https://github.com/pytorch/pytorch/blob/99a423f8fce967169a1bf8d87cf2d79d1d80087b/torch/csrc/utils/tensor_new.cpp#L224
- re supported objects other than the mentioned ones: yes, these are objects that implement PEP 3118 Buffer protocol (memoryview) but it was not mentioned here because I think the support is broken in the sense that `as_tensor(<memoryview>)` unnecessarily copies data, see p.4 in https://pearu.github.io/array_interface_pytorch.html#recommendations-1 . But this is another issue..

Other objects that could be supported in `torch.as_tensor/tensor` are:
- pyarrow.Array and CudaBuffer
- pandas.Series
- cudf.Series
- objects implementing DLPack interface
- etc
but it might be easier to implement the CUDA/CPU Array Interfaces in the above mentioned objects rather than providing the direct support with PyTorch.